packages/expo-battery/ios/EXBattery/EXBattery.m

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXBattery/EXBattery.h>
+#import <UMCore/UMUtilities.h>
+
+@interface EXBattery ()
+
+@property (nonatomic, weak) UMModuleRegistry *moduleRegistry;
+@property (nonatomic, weak) id <UMEventEmitterService> eventEmitter;
+@property (nonatomic, assign) BOOL hasListeners;
+
+@end
+
+@implementation EXBattery
+
+UM_EXPORT_MODULE(ExpoBattery);
+  
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  if (_moduleRegistry) {
+    [self invalidate];
+  }
+  _moduleRegistry = moduleRegistry;
+  _eventEmitter = [moduleRegistry getModuleImplementingProtocol:@protocol(UMEventEmitterService)];
+  
+  if (moduleRegistry) {
+    [UIDevice.currentDevice setBatteryMonitoringEnabled:YES];
```

Comment:
are there ramifications of setting this to YES that we should document?

Reply:
I didn't find any specific side effects or ramifications. The official docs state that 
```
Enable battery monitoring if your app needs to be notified of changes to the battery state, or if you want to check the battery charge level.

The default value of this property is NO, which:

Disables the posting of battery-related notifications

Disables the ability to read battery charge level and battery state
```
But I do see in our implementation that when in the uni-module lifecycle, when our `expo-battery` is registered in `setModuleRegistry`, event listeners are also registered for `NSNotificationsCenter`. Is this the desired behavior? From what I understand, perhaps we would only want to register these event listeners when the user chooses to subscribe?

Reply:
cool, thanks for researching this thoroughly. Yes, I think you're right - we probably want to add these observers in the `startObserving` method and then remove them in `stopObserving`. I believe RN takes care of calling those methods when the first listener is added and when the last listener is removed in JS.